internal/infrastructure/ai_client_test.go

```
@@ -0,0 +1,56 @@
+
+func TestAIClient_GenerateReply_Success(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "/ai/reply", r.URL.Path)
+		assert.Equal(t, http.MethodPost, r.Method)
+
+		var body map[string]any
+		assert.NoError(t, json.NewDecoder(r.Body).Decode(&body))
+		assert.Equal(t, "promo masih ada?", body["customer_message"])
+		assert.Equal(t, "628123", body["phone_number"])
+
+		json.NewEncoder(w).Encode(domain.AIReplyResponse{
```

golangci-lint (2.12.2)

[error] Error return value of `(*encoding/json.Encoder).Encode` is not checked

(errcheck)